src/core/ngx_inet.c

```diff
@@ -972,13 +1285,25 @@ ngx_inet_resolve_host(ngx_pool_t *pool, ngx_url_t *u)
 #endif
 
     if (getaddrinfo((char *) host, NULL, &hints, &res) != 0) {
+#if (T_NGX_DNS_RESOLVE_BACKUP)        
+        ph = (u_char *) getenv(NGX_DNS_RESOLVE_BACKUP_PATH);
+        if (ph != NULL){
+            path.data = ph;
+            path.len = ngx_strlen(ph);
+            if (ngx_resolve_using_local(pool, &u, path) == NGX_OK) {
+                ngx_log_error(NGX_LOG_WARN, ngx_cycle->log, 0, 
+                              "dom %V using local dns cache successed",
+                              &u->host);                
+                ngx_free(host);
+                return NGX_OK;
+            }
+        }
+#endif   
         u->err = "host not found";
         ngx_free(host);
         return NGX_ERROR;
     }
 
-    ngx_free(host);
-
```

Comment:
Should we move ngx_free(host) to end of function?

not found host used in T_NGX_DNS_RESOLVE_BACKUP.

Reply:
Oh, Now ngx_free(host) doesn't have to be at the end of function.   It's has been updated, PLZ Review Thx.